crypto/fipsmodule/rand/new_rand.c

```diff
@@ -50,10 +54,27 @@ static int rand_ensure_valid_state(void) {
   return 1;
 }
 
-// TODO
-// For UBE.
-static int rand_ensure_ctr_drbg_uniquness(struct rand_thread_local_state *state,
-  size_t out_len) {
+// rand_ensure_ctr_drbg_uniquness computes whether |state| must be randomized to
+// ensure uniqueness.
+//
+// Note: If |rand_ensure_ctr_drbg_uniquness| returns 0 it does not necessarily
+// imply that an UBE occurred. It can also mean that no UBE detection is
+// supported or that UBE detection failed. In these cases, |state| must also be
+// randomized to ensure uniqueness. Any special future cases can be handled in
+// this function. 
```

Comment:
How can the caller determine if 0 means no UBE (do not randomize state), or if 0 means no UBE supported (must randomize state)? Would returning 1 be safer, so callers know if this returns 1 they must randomize regardless of reason.

Reply:
My bad, it's a typo. The `0` in line 60 should be a 1...